includes/manage_collection.inc

```diff
@@ -386,6 +386,46 @@ function islandora_basic_collection_share_children_form(array $form, array &$for
 }
 
 /**
+ * Form validation for share children form.
+ *
+ * @param array $form
+ *    The Drupal form definition.
+ * @param array $form_state
+ *    The Drupal form state.
+ */
+function islandora_basic_collection_share_children_form_validate(array $form, array &$form_state) {
+  $new_collection = $form_state['values']['collection'];
+  $clicked_button = end($form_state['clicked_button']['#parents']);
+  $get_pid = function($o) {
+    if (isset($o['object']['value'])) {
+      return $o['object']['value'];
+    }
+    else {
+      return '';
+    }
+  };
+
+  if ($clicked_button == 'submit_all') {
```

Comment:
Huh yeah, guess thats what happens when I'm up too late. I'll push this to a single function and call it from both.